.github/workflows/build.yml

```diff
@@ -61,126 +25,110 @@ jobs:
         || github.ref == 'refs/heads/master'
         || !contains(github.event.head_commit.message, '[ci only windows]')
+        || !contains(github.event.head_commit.message, '[ci only macos]')
       )
 
     env:
-      CCACHE_DIR: ${{ github.workspace }}/.ccache
+      CCACHE_DIR: ${{ github.workspace }}/.cache/ccache
       CCACHE_COMPRESS: "1"
-      PY3_PKG: python-3.7.6-macosx10.9.pkg
-      PY3_URL: https://www.python.org/ftp/python/3.7.6/python-3.7.6-macosx10.9.pkg
-      HOMEBREW_CACHE: ${{ github.workspace }}/.cache/brew
-      HOMEBREW_NO_INSTALL_CLEANUP: "1"
-    services: ${{ matrix.platform.services }}
+      PY_VER: "3.7"
+    services:
+      postgis:
+        image: postgis/postgis
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 2s
+          --health-timeout 2s
+          --health-retries 5
+          -e POSTGRES_HOST_AUTH_METHOD=trust
+        ports:
+          - 5432:5432
+      sqlserver:
+        image: mcr.microsoft.com/mssql/server
+        options: >-
+          -e ACCEPT_EULA=Y
+          -e SA_PASSWORD=PassWord1
+        ports:
+          - 1433:1433
+      mysql:
+        image: mysql
+        options: >-
+          -e MYSQL_ROOT_PASSWORD=PassWord1
+        ports:
+          - 3306:3306
+
     steps:
       - uses: actions/checkout@v2
 
-      #
-      # vendor source
-      #
-
-      - name: "vendor-source: cache"
-        id: cache-vendor
-        uses: actions/cache@v1
-        with:
-          path: .cache/vendor-source
-          key: vendor-source-${{ matrix.platform.id }}-${{ hashFiles('vendor/**/Makefile') }}-${{ hashFiles('vendor/Makefile') }}
-
-      - name: "vendor-source: download"
-        if: steps.cache-vendor.outputs.cache-hit != 'true'
-        run: |
-          tar xvf .cache/vendor-source/vendor.tar || true
-          echo 'verbose = off' >> $HOME/.wgetrc
-          make -C vendor sources
-          mkdir -p .cache/vendor-source
-          tar cvf .cache/vendor-source/vendor.tar vendor/*/*.tar.* vendor/*/*.zip
-
       #
       # python distribution
       #
 
-      - name: "🍎 python: cache"
-        id: cache-pydist
-        uses: actions/cache@v1
-        if: runner.os == 'macOS'
-        with:
-          path: .cache/pydist
-          key: pydist-${{ matrix.platform.id}}-${{ env.PY3_PKG }}
-
-      - name: "🍎 python: download"
-        if: runner.os == 'macOS' && steps.cache-pydist.outputs.cache-hit != 'true'
-        run: |
-          mkdir -p .cache/pydist
-          wget -nv ${{ env.PY3_URL }} -O .cache/pydist/${{ env.PY3_PKG }}
-
       - name: "python: pip cache"
         uses: actions/cache@v1
         with:
           path: .cache/pip
-          key: pip-${{ runner.os }}-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements/*.txt') }}-${{ hashFiles('Makefile') }}
+          key: pip-Linux-${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements/*.txt') }}-${{ hashFiles('Makefile') }}
```

Comment:
FYI hashFiles has been updated and now takes multiple glob arguments so that:
`${{ hashFiles('requirements.txt') }}-${{ hashFiles('requirements/*.txt') }}`
is equivalent in some sense to:
`${{ hashFiles('requirements.txt', 'requirements/*.txt') }}`
Lots more opportunities in this file to do this.
This could be as good a time as any to tidy that up, although note that it will effectively clear all caches.

Reply:
ah, ace. Willdo.